clients/src/test/java/org/apache/kafka/common/config/provider/EnvVarConfigProviderTest.java

```diff
@@ -0,0 +1,104 @@
+import java.util.Arrays;
+import java.util.Set;
+import java.util.Map;
+import java.util.HashSet;
+import java.util.HashMap;
+
+import static org.apache.kafka.common.config.provider.EnvVarConfigProvider.ENV_VAR_CONFIG_PROVIDER_PATTERN_CONFIG;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+class EnvVarConfigProviderTest {
+
+    private EnvVarConfigProvider envVarConfigProvider = null;
+    @BeforeEach
+    public void setup() {
+        Map<String, String> testEnvVars = new HashMap<String, String>() {
+            {
+                put("test_var1", "value1");
+                put("secret_var2", "value2");
+                put("new_var3", "value3");
+            }
+        };
+        envVarConfigProvider = new EnvVarConfigProvider(testEnvVars);
+        envVarConfigProvider.configure(Collections.singletonMap("", ""));
+    }
+
+    @Test
+    void testGetAllEnvVarsNotEmpty() {
+        ConfigData properties = envVarConfigProvider.get("");
+        assertNotEquals(0, properties.data().size());
+    }
+
+    @Test
+    void testGetMultipleKeysAndCompare() {
+        ConfigData properties = envVarConfigProvider.get("");
+        assertNotEquals(0, properties.data().size());
+        assertEquals("value1", properties.data().get("test_var1"));
+        assertEquals("value2", properties.data().get("secret_var2"));
+        assertEquals("value3", properties.data().get("new_var3"));
+    }
+
+    @Test
+    public void testGetOneKeyWithNullPath() {
+        ConfigData config = envVarConfigProvider.get(null, Collections.singleton("secret_var2"));
+        Map<String, String> data = config.data();
+
+        assertEquals(1, data.size());
+        assertEquals("value2", data.get("secret_var2"));
+    }
+
+    @Test
+    public void testGetOneKeyWithEmptyPath() {
+        ConfigData config = envVarConfigProvider.get("", Collections.singleton("test_var1"));
+        Map<String, String> data = config.data();
+
+        assertEquals(1, data.size());
+        assertEquals("value1", data.get("test_var1"));
+    }
+
+    @Test
+    void testGetWhitelistedEnvVars() {
+        Set<String> whiteList = new HashSet<>(Arrays.asList("test_var1", "secret_var2"));
+        Set<String> keys = envVarConfigProvider.get(null, whiteList).data().keySet();
+        assertEquals(whiteList, keys);
+    }
+    @Test
+    void testNotNullPathNonEmptyThrowsException() {
+        assertThrows(ConfigException.class, () -> envVarConfigProvider.get("test-path", Collections.singleton("test_var1")));
+    }
+
+    @Test void testRegExpEnvVars() {
+        Map<String, String> testConfigMap = Collections.singletonMap(ENV_VAR_CONFIG_PROVIDER_PATTERN_CONFIG, "secret_.*");
+        envVarConfigProvider.configure(testConfigMap);
+
+        assertEquals(1, envVarConfigProvider.get(null, Collections.singleton("secret_var2")).data().size());
```

Comment:
Does this also pass for `custom_secret_var2`?

In other words, I believe this is testing `find(pattern)`, not strict `matches`, which has implicit `^ $` anchors. 

There seems to be a difference between `asPredicate` and `asMatchPredicate`. 

https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/util/regex/Pattern.html#asPredicate()

Reply:
Of course - thanks for pointing that out. It doesn't pass because I am using the `asPredicate` method. Added a more detailed test case but not sure about `asMatchPredicate` because ist not Java 8 compatible (was added in 11). It's added and tested locally but not yet pushed. Might need to use the matcher instead to stay compatible.